src/executor.rs

```diff
@@ -0,0 +1,98 @@
+// FIXME: `expect`s should be removed and proper errors returned.
+use std::convert::TryFrom;
+use std::pin::Pin;
+use std::sync::Arc;
+
+use futures::{
+    future::Future,
+    task::{Context, Poll},
+};
+use libconsensus_lachesis_rs::tcp_server::{TcpNode, TcpPeer};
+use libconsensus_lachesis_rs::{BTreeHashgraph, Event, Node, Swirlds};
+use log::debug;
+use vm::instruction::Program;
+use vm::Cpu;
+
+/// The new programs returned by the node.
+struct NewPrograms {
+    /// The node that produces programs.
+    node: Arc<TcpNode<Swirlds<TcpPeer, BTreeHashgraph>>>,
+    /// The index of the next program.
+    program_index: usize,
+}
+
+impl Future for NewPrograms {
+    type Output = Vec<Program>;
+
+    fn poll(self: Pin<&mut Self>, _cxt: &mut Context) -> Poll<Self::Output> {
```

Comment:
This is non-conventional implementation of the poll() method for Future trait, it misses to take waker out of Context and calling it when ready to be polled again.